client-http/src/main/java/com/cloudera/livy/client/http/LivyConnection.java

```diff
@@ -75,8 +80,32 @@ public int getConnectTimeout() {
       public int getSocketTimeout() {
         return (int) config.getTimeAsMs(SOCKET_TIMEOUT);
       }
+
+      @Override
+      public boolean isAuthenticationEnabled() {
+        return true;
+      }
+    };
+
+    Credentials dummyCredentials = new Credentials() {
+      @Override
+      public String getPassword() {
+        return null;
+      }
+
+      @Override
+      public Principal getUserPrincipal() {
+        return null;
+      }
     };
 
+    // This is needed to get Kerberos credentials from the environment, instead of
+    // requiring the application to manually obtain the credentials.
+    System.setProperty("javax.security.auth.useSubjectCredsOnly", "false");
```

Comment:
The HTTPClient uses JAAS (Java's auth framework) to get credentials. This tells JAAS that it's ok to read credentials from the environment (e.g. the ticket cache in `/tmp/krb5cc_0`).